src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -479,9 +482,15 @@ ModifyQuerySupported(Query *queryTree)
 	 */
 	if (queryTree->hasSubLinks == true)
 	{
-		return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
-							 "subqueries are not supported in distributed modifications",
-							 NULL, NULL);
+		/* we have an exception for UPDATE queries if multiShardQuery is set false */
+		if ((commandType != CMD_UPDATE && commandType != CMD_DELETE) || multiShardQuery)
```

Comment:
Could maybe introduce an `IsUpdateOrDelete(queryTree)` function for cleanliness.